### PR TITLE
User story 10

### LINKED ISSUE #10: User Story 10, User Registration
As a visitor
When I click on the 'register' link in the nav bar
Then I am on the user registration page ('/register')
And I see a form where I input the following data:
- my name
- my street address
- my city
- my state
- my zip code
- my email address
- my preferred password
- a confirmation field for my password

When I fill in this form completely,
And with a unique email address not already in the system
My details are saved in the database
Then I am logged in as a registered user
I am taken to my profile page ("/profile")
I see a flash message indicating that I am now registered and logged in

### PR DESCRIPTION
**Pull request by dhalverson**
Adds test for form and nav bar
Adds link to view
Adds users controller and actions

Need to create table for user data still
Need to finalize create action on users controller

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class SessionsController < ApplicationController
```
Comment (PhilipDeFraties) [inline app/controllers/sessions_controller.rb]: I think this file should actually be removed for this pull request as its not needed until user story 13

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+class UsersController < ApplicationController
+
+  def new
+  end
+
+  def create
+    user = User.new(user_params)
+    if user.save
```
Comment (PhilipDeFraties) [inline app/controllers/users_controller.rb]: Technically I think that setting up the conditional here is outside what is needed for this user story, it should have just been user.save, then implement the conditional with sad path for user story 12

**Review by PhilipDeFraties — CHANGES_REQUESTED** [review]
I think the sessions_controller file should be removed as it was mistakenly implemented, other than that I think the conditional in the user create action could be removed as its outside the scope of this user story but maybe not super important.

File: app/controllers/sessions_controller.rb (hunk 1 above)
Comment (dhalverson) [reply app/controllers/sessions_controller.rb]: Done!

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (dhalverson) [reply app/controllers/users_controller.rb]: Removed as well

**Review by dhalverson — COMMENTED** [review]
These changes have been made

**Comment by D-Lessenden** [comment]
Changes requested by Phil have been made.

**Pull request merged**